src/Vision/V1/ImageAnnotatorClient.php

```diff
@@ -30,13 +30,458 @@
+     *
+     *     @type ImageContext        $imageContext  Additional context that may accompany the image.
+     *     @type RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return AnnotateImageResponse
+     *
+     * @throws ApiException if the remote call fails
+     * @experimental
+     */
+    public function annotateImage($image, $features, $optionalArgs = [])
+    {
+        return $this->annotateImageHelper(
+            [$this, 'batchAnnotateImages'],
+            AnnotateImageRequest::class,
+            $image,
+            $features,
+            $optionalArgs
+        );
+    }
+
+    /**
+     * Run face detection for an image.
+     *
+     * Example:
+     * ```
+     * $imageContent = file_get_contents('path/to/image.jpg');
+     * $image = $imageAnnotatorClient->createImageObject($imageContent);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * @param Image $image        An image annotation request.
+     * @param array $optionalArgs {
+     *                            Optional.
+     *
+     *     @type RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return AnnotateImageResponse
+     *
+     * @throws ApiException if the remote call fails
+     * @experimental
+     */
+    public function faceDetection($image, $optionalArgs = [])
+    {
+        return $this->annotateSingleFeature(
+            $image,
+            Feature_Type::FACE_DETECTION,
+            $optionalArgs
+        );
+    }
+
+    /**
+     * Run landmark detection for an image.
+     *
+     * Example:
+     * ```
+     * $imageContent = file_get_contents('path/to/image.jpg');
+     * $image = $imageAnnotatorClient->createImageObject($imageContent);
+     * $response = $imageAnnotatorClient->landmarkDetection($image);
+     * ```
+     *
+     * @param Image $image        An image annotation request.
+     * @param array $optionalArgs {
+     *                            Optional.
+     *
+     *     @type RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return AnnotateImageResponse
```

Comment:
WDYT about keying directly in to the result users are interested in and returning that instead (e.g. `$imageResponse->getLandmarkAnnotations()`)? Taking a look at `AnnotateImageResponse` the only other field of interest if a user specifies one of these features through this helper is the `error`. In the case we do get an error, we could bubble up an exception - so that users don't need to check the field themselves on each request.

Reply:
I would worry about changing `error` into an exception. That is a very different model to the existing `BatchAnnotateImages` method. Also, it seems like the error field can contain useful information even if info is filled in the response: https://github.com/googleapis/googleapis/blob/master/google/cloud/vision/v1/image_annotator.proto#L564

Reply:
That sounds fair.